.github/workflows/ci.yml

```diff
@@ -14,27 +14,29 @@ jobs:
         shell: bash
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-12, windows-latest]
+        os:
+          - ubuntu-latest
+          - macos-13
+          # libfaketime tests fail on macos arm. Disable tests for now.
+          # - macos-14
+          - windows-latest
         python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
       fail-fast: false
     env:
-      # 11.7 necessary due to: https://github.com/actions/setup-python/issues/682#issuecomment-1604261330
-      MACOSX_DEPLOYMENT_TARGET: ${{ matrix.os == 'macos-12' && matrix.python-version == '3.8' && '11.7' || '11' }}
-      #MACOSX_DEPLOYMENT_TARGET: "10.11"
-      # On windows-2019 we are using the Visual Studio generator, which is multi-config and places the build artifacts in a subdirectory
+      MACOSX_DEPLOYMENT_TARGET: "11"
     steps:
       - name: Checkout TileDB-Py `dev`
         uses: actions/checkout@v3
```

Comment:
we should use `actions/checkout@v4` wherever it appears: https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/